django_netjsonconfig/tests/test_device.py

```diff
@@ -68,6 +68,28 @@ def test_backend(self):
         self.assertIsNotNone(d.backend)
         self.assertEqual(d.backend, c.get_backend_display())
 
+    def test_add_device_with_context(self):
+        d = self._create_device()
+        d.additional_context = {
+            'key': 'thisIsAddedContext'
+        }
+        d.save()
+        c = self._create_config(device=d, config={
+            "openwisp": [
+                {
+                    "config_name": "controller",
+                    "config_value": "http",
+                    "url": "http://controller.examplewifiservice.com",
+                    "interval": "60",
+                    "verify_ssl": "1",
+                    "uuid": "UUID",
+                    "key": "{{ key }}"
+                }
+            ]
+        })
+        self.assertEqual(c.json(dict=True)['openwisp'][0]['key'],
+                         'thisIsAddedContext')
```

Comment:
@nemesisdesign, what do you mean by "using the interval"?

Reply:
setting a context variable for `interval`, because `key` is already set by default by code that was already existing before: https://github.com/openwisp/django-netjsonconfig/blob/master/django_netjsonconfig/base/config.py#L95